search-api/search_api/resources/directors.py

```diff
@@ -0,0 +1,215 @@
+            # CorpParty.party_typ_cd
+            _ = sheet.cell(column=7, row=index, value=row.party_typ_cd)
+            # CorpParty.appointment_dt
+            _ = sheet.cell(column=8, row=index, value=row.appointment_dt)
+            # CorpParty.cessation_dt
+            _ = sheet.cell(column=9, row=index, value=row.cessation_dt)
+            # CorpOpState.state_typ_cd
+            _ = sheet.cell(column=10, row=index, value=_get_state_typ_cd_display_value(row.state_typ_cd))
+            # CorpName.corp_nme
+            _ = sheet.cell(column=11, row=index, value=row.corp_nme)
+            # Corporation.corp_num
+            _ = sheet.cell(column=12, row=index, value=row.corp_num)
+
+        current_date = datetime.datetime.strftime(datetime.datetime.now(), "%Y-%m-%d %H:%M:%S")
+        filename = "Director Search Results {date}.xlsx".format(date=current_date)
+        full_filename_path = "{dir}/{filename}".format(dir=export_dir, filename=filename)
+        wb.save(filename=full_filename_path)
+
+        return send_from_directory(export_dir, filename, as_attachment=True)
+
+
+@API.route('/<id>')
+@jwt.requires_auth
+def person(id):
+    result = CorpParty.get_corporation_info_by_corp_party_id(id)
+
+    person = result[0]
+    result_dict = {}
+
+    filing_description = CorpParty.get_filing_description_by_corp_party_id(id)
+
+    name = CorpName.get_corp_name_by_corp_id(person.corp_num)[0]
+    offices = Office.get_offices_by_corp_id(person.corp_num)
+    delivery_addr = _normalize_addr(person.delivery_addr_id)
+    mailing_addr = _normalize_addr(person.mailing_addr_id)
+
+    states = CorpState.get_corp_states_by_corp_id(person.corp_num)
+
+    corp_delivery_addr = ';'.join([_normalize_addr(office.delivery_addr_id) for office in offices])
+    corp_mailing_addr = ';'.join([_normalize_addr(office.mailing_addr_id) for office in offices])
+
+    result_dict['corp_party_id'] = int(person.corp_party_id)
+    result_dict['first_nme'] = person.first_nme
+    result_dict['middle_nme'] = person.middle_nme
+    result_dict['last_nme'] = person.last_nme
+    result_dict['appointment_dt'] = person.appointment_dt
+    result_dict['cessation_dt'] = person.cessation_dt
+    result_dict['corp_num'] = person.corp_num
+    result_dict['corp_nme'] = name.corp_nme
+    result_dict['party_typ_cd'] = person.party_typ_cd
+    result_dict['corp_party_email'] = person.email_address
+    result_dict['delivery_addr'] = delivery_addr
+    result_dict['mailing_addr'] = mailing_addr
+    result_dict['corp_delivery_addr'] = corp_delivery_addr
+    result_dict['corp_mailing_addr'] = corp_mailing_addr
+    result_dict['corp_typ_cd'] = result.corp_typ_cd
+    result_dict['corp_admin_email'] = result.admin_email
+    result_dict['full_desc'] = filing_description[0].full_desc if filing_description else None
+
+    result_dict['states'] = [s.as_dict() for s in states]
+
+    return jsonify(result_dict)
+
+
+@API.route('/officesheld/<corppartyid>')
```

Comment:
I think what we are finding here are the offices held by the director. If so, the endpoint could be  /directors/<party_id>/offices ?
officesheld is not a resource